src/Toml/Type/AnyValue.hs

```diff
@@ -30,13 +30,7 @@ instance Show AnyValue where
     show (AnyValue v) = show v
 
 instance Eq AnyValue where
-    (AnyValue (Bool b1))    == (AnyValue (Bool b2))    = b1 == b2
-    (AnyValue (Integer i1)) == (AnyValue (Integer i2)) = i1 == i2
-    (AnyValue (Double f1))  == (AnyValue (Double f2))  = isRight $ sameValue (Double f1) (Double f2)
-    (AnyValue (Text s1))    == (AnyValue (Text s2))    = s1 == s2
-    (AnyValue (Date d1))    == (AnyValue (Date d2))    = d1 == d2
-    (AnyValue (Array a1))   == (AnyValue (Array a2))   = eqValueList a1 a2
-    _                       == _                       = False
+    (AnyValue val1)  == (AnyValue val2)  = isRight $ sameValue val1 val2
```

Comment:
It's not enough to call `isRight` on `sameValue`. You also need to do `val1 == val2` here.